### PR TITLE
Fix incorrect unit test directory in Gradle project and failing test

### PR DESCRIPTION
**Pull request by ghost**
#### As discussed in IRC:

Gradle/Maven expects unit tests to be found at `src/test/java`, however SpongeForge has been using `src/main/test/java`. This is incorrect.

---

Appended class with `@PowerMockIgnore` annotation to prevent PowerMockito from mocking classes we don't want to mock.

Also adds another `EventManager#registerListeners()` call in `testSpawn()`, as this **_should_** be expected behavior (we don't want to fire an event if nothing is listening for it - wasted processing time).

### CODE REVIEW

**Comment by jamierocks** [comment]
The directory is `org.spongepowered.mod` it should be `org/spongepowered/mod`.

**Comment by ghost** [comment]
Oops. Looks like someone who created the test before me had this as the directory name, and my commit just continues on that mistake. Never mind; I'll fix.

File: src/test/java/org.spongepowered.mod/event/SpongeModEventManagerTest.java [hunk 1]
```diff
@@ -75,6 +96,7 @@ public void testSpawn() throws ClassNotFoundException {
 
         eventBus.register(forgeListener);
 
+        this.eventManager.registerListeners(this.plugin, listener);
```
Comment (JBYoshi) [inline src/test/java/org.spongepowered.mod/event/SpongeModEventManagerTest.java:99]: I don't think this is supposed to be here. This section of the test is making sure that the fields are `true` when only Forge listeners are registered.

**Review by JBYoshi — CHANGES_REQUESTED** [review]
I was looking at this earlier. I don't think this test works any more. Most of the test can actually be moved to SpongeCommon; only the Forge event section needs to stay here.

**EDIT:** The Forge-independent part is already in SpongeCommon (`ShouldFireTest`).

**Comment by ghost** [comment]
Quite possibly not. Been well over a year since I submitted this merge request. I'm still interested in supporting it. If I get time, I'll update the branch - has since been deleted from my local repo.

File: src/test/java/org.spongepowered.mod/event/SpongeModEventManagerTest.java (hunk 1 above)
Comment (ghost) [reply src/test/java/org.spongepowered.mod/event/SpongeModEventManagerTest.java:99]: Original MR message stated

    Also adds another EventManager#registerListeners() call in testSpawn(), 
    as this should be expected behavior (we don't want to fire an event if
    nothing is listening for it - wasted processing time)."

You could be right, though. Could we have someone else's thoughts?

**Comment by ghost** [comment]
See #1896 for an update, since I deleted the original repo in which this MR was from

**Pull request closed without merging**